### PR TITLE
fix a bug: when use tf.keras.layers.TextVectorization layer to load model

### PR DESCRIPTION
**Pull request by mikuh**
here need to compare strings for equality not string memory address

### CODE REVIEW

**Comment by mikuh** [comment]
@googlebot I signed it!

File: keras/layers/preprocessing/text_vectorization.py [hunk 1]
```diff
@@ -513,7 +513,7 @@ def call(self, inputs):
     lookup_data = self._lookup_layer(inputs)
 
     # For any non-int output, we can return directly from the underlying layer.
-    if self._output_mode is not INT:
+    if self._output_mode != INT:
```
Comment (fchollet) [inline keras/layers/preprocessing/text_vectorization.py:516]: Both are equivalent (the objects being compared aren't pointers)

Comment (mikuh) [reply keras/layers/preprocessing/text_vectorization.py:516]: see this issue: https://github.com/keras-team/keras/issues/15382

**Review by fchollet — APPROVED** [review]
LGTM

**Pull request merged**